terraform/platforms/aws/config.tf

```diff
@@ -0,0 +1,38 @@
+// This bucket is used for saving the ignition config
+resource "aws_s3_bucket" "config" {
```

Comment:
Nit thought here, what do you think? -->
- Rename this whole file `s3.tf`
- Could move `resource "aws_s3_bucket" "backups" {` from `aws.tf` in here too
- Add the following configuration to the `backups` bucket to get server-side encryption with the KMS key - obviously, it may be better to do like you do for the ignition config & encrypt it locally (i.e. on the etcd nodes themselves), but would require some additional Go changes.

```
  server_side_encryption_configuration {
    rule {
      apply_server_side_encryption_by_default {
        kms_master_key_id = "${aws_kms_key.mykey.arn}"
        sse_algorithm     = "aws:kms"
      }
    }
```

Reply:
With this in place, will the `aws_s3_bucket_object` still need the kms encryption block, or does this rule just enforce it?

Will change the filename

Reply:
If you add this block to the `backups` bucket, it should get S3 to automatically server-side encrypt the backups. ECO would still just send them plain, but over TLS (except if we make a Go code change to make it encrypt them with KMS client-side too). 

However, my understanding (would need to look in the Terraform code) is that by adding `kms_key_id` to the `aws_s3_bucket_object" for the ignition config, Terraform should client-side encrypt the content & upload it as is?

But I am not so so familiar with KMS to be honest so take it with a grain of salt..

Reply:
^ Whatever you do / decide is reasonably fine by me honestly, pick what you feel comfortable with.